web-marketplace/src/pages/ProfileEdit/ProfileEdit.Settings.tsx

```diff
@@ -0,0 +1,98 @@
+import { useState } from 'react';
+
+import ErrorBanner from 'web-components/lib/components/banner/ErrorBanner';
+
+import { useAuth } from 'lib/auth/auth';
+import { apiServerUrl } from 'lib/env';
+import { useSignArbitrary } from 'lib/wallet/hooks/useSignArbitrary';
+import { useWallet } from 'lib/wallet/wallet';
+
+import { AccountConnectWalletModal } from 'components/organisms/AccountConnectWalletModal/AccountConnectWalletModal';
+import { useLoginData } from 'components/organisms/LoginButton/hooks/useLoginData';
+import { UserAccountSettings } from 'components/organisms/UserAccountSettings/UserAccountSettings';
+import {
+  SocialProviderInfo,
+  WalletProviderInfo,
+} from 'components/organisms/UserAccountSettings/UserAccountSettings.types';
+
+import { useConnectKeplrWallet } from './hooks/useConnectKeplrWallet';
+
+export const ProfileEditSettings = () => {
+  const [error, setError] = useState<unknown>(undefined);
+  const { authenticatedAccounts, activeAccount } = useAuth();
+  const { wallet } = useWallet();
+
+  const {
+    connecting,
+    isModalOpen,
+    modalState,
+    onButtonClick,
+    onModalClose,
+    qrCodeUri,
+    walletsUiConfig,
+  } = useLoginData();
+
+  const signArbitrary = useSignArbitrary({
+    setError,
+  });
+
+  // Google accounts
+  const googleAccounts = authenticatedAccounts?.filter(
+    account => !!account?.email,
+  );
+
+  const googleSocialProviders: SocialProviderInfo[] =
+    googleAccounts?.map(account => ({
+      providerName: `Google`,
+      address: `${account?.email}`,
+      disconnect: () => undefined,
+    })) ?? [];
+  const connectGoogleAccount = () =>
+    (window.location.href = `${apiServerUrl}/marketplace/v1/auth/google`);
+
+  // Keplr accounts
+  const hasKeplrAccount = !!activeAccount?.addr;
+  const isCurrrentAddressAuthenticated =
+    authenticatedAccounts?.some(account => account?.addr === wallet?.address) ??
+    false;
+  const walletProviderInfo: WalletProviderInfo = hasKeplrAccount
+    ? { address: String(activeAccount?.addr), disconnect: () => void 0 }
+    : { connect: onButtonClick };
+
+  useConnectKeplrWallet({
+    setError,
+    signArbitrary,
+    hasKeplrAccount,
+    isCurrrentAddressAuthenticated,
+  });
+
+  return (
+    <>
+      {error && (
+        <ErrorBanner
+          text={error.toString()}
+          onClose={() => setError(undefined)}
+        />
+      )}
+      <UserAccountSettings
+        email={activeAccount?.email ?? ''}
+        socialProviders={[
+          {
+            providerName: 'Google',
+            connect: connectGoogleAccount,
+          },
+          ...googleSocialProviders,
```

Comment:
if connected to google, I'm seeing this, it should show either the connected or disconnected state:

<img width="848" alt="image" src="https://github.com/regen-network/regen-web/assets/3149358/f1499aba-3090-472d-8a24-49f3b756db3b">

Reply:
FYI @flagrede I've fixed this in https://github.com/regen-network/regen-web/pull/2206/commits/665f1d98971d03e4ac57dae06564cb99dc36a66b
We only want to display the connected google account for the current active account.